pkg/metrics/server.go

```diff
@@ -0,0 +1,99 @@
+package metrics
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/spf13/cobra"
+	"k8s.io/klog"
+)
+
+type Server struct {
+	http             http.Server
+	shutdownFinished chan struct{}
+}
+
+func (server *Server) ListenAndServe() {
+	err := server.http.ListenAndServe()
+
+	if errors.Is(err, http.ErrServerClosed) {
```

Comment:
I think you can just do `err == http.ErrServerClosed` or `err != http.ErrServerClosed`. This is what is given as example from `net/http`.